pkg/asset/installconfig/alibabacloud/validation.go

```diff
@@ -0,0 +1,85 @@
+package alibabacloud
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/openshift/installer/pkg/types"
+)
+
+// Validate executes platform-specific validation.
+func Validate(client *Client, ic *types.InstallConfig) error {
+	allErrs := field.ErrorList{}
+	platformPath := field.NewPath("platform").Child("alibabacloud")
+	allErrs = append(allErrs, validatePlatform(client, ic, platformPath)...)
+
+	return allErrs.ToAggregate()
+}
+
+func validatePlatform(client *Client, ic *types.InstallConfig, path *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if ic.Platform.AlibabaCloud.ResourceGroupID != "" {
+		allErrs = append(allErrs, validateResourceGroup(client, ic, path)...)
+	}
+	return allErrs
+}
+
+func validateResourceGroup(client *Client, ic *types.InstallConfig, path *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if ic.AlibabaCloud.ResourceGroupID == "" {
+		return allErrs
```

Comment:
This should return an error when the resource group ID is missing? Moreover, static validation (ie, validation that does not require a connection to the cloud provider), should be done in `pkg/types/alibabacloud/validation`.

Reply:
Yes, currently in our design, ReourceGroupID is a required parameter. I moved this part of the validation to `pkg/types/alibabacloud/validation`.